dcctools/config.py

```diff
@@ -124,6 +168,29 @@ def extract_blocks(self, system):
         with ZipFile(block_zip_path, mode="r") as block_zip:
             block_zip.extractall(Path(self.config_json["inbox_folder"]) / system)
 
+    def get_metadata(self, system):
+        archive_name = (
+            f"{system}_households.zip" if self.household_match else f"{system}.zip"
+        )
+        archive_path = Path(self.config_json["inbox_folder"]) / archive_name
+        with ZipFile(archive_path, mode="r") as archive:
+            for file_name in archive.namelist():
+                if "metadata" in file_name:
+                    with archive.open(file_name) as metadata_file:
+                        return json.load(metadata_file)
+
+    def get_clk_path(self, system, project):
+        archive_name = (
+            f"{system}_households.zip" if self.household_match else f"{system}.zip"
+        )
+        clk_zip_path = Path(self.config_json["inbox_folder"]) / archive_name
+        with ZipFile(clk_zip_path, mode="r") as clk_zip:
+            target = f"households/{project}" if self.household_match else project
+            for file_name in clk_zip.namelist():
+                if target in file_name:
+                    if len(file_name) == len(target) + 21:
```

Comment:
What's the `+ 21`? I'm guessing this might have to do with the error I see below.

Reply:
the filename of a given clk `.json` should be 21 characters longer than the length of the name of the project it corresponds to. It's a way to ensure that if we were hypothetically looking for `name-sex-phone` we don't also return `name-sex-dob-phone` (which contains the former as a substring)

Reply:
Can you provide an example? I still don't understand the project name breakdown.

e.g. from the [below example](https://github.com/mitre/linkage-agent-tools/pull/29#discussion_r977799559) the project name is `output/name-sex-dob-phone.json` and the file_name is `output/name-sex-dob-zip.json`. This appears to be based on the assumption that the project CLK files have timestamps in the name, which is not the case.